sips/sip-6.md

```diff
@@ -0,0 +1,435 @@
+
+```Rust
+/// A self-custodial object holding the staked SUI tokens.
+struct StakedSui has key {
+    id: UID,
+    /// ID of the staking pool we are staking with.
+    pool_id: ID,
+    /// The epoch at which the stake becomes active.
+    stake_activation_epoch: u64,
+    /// The staked SUI tokens.
+    principal: Balance<SUI>,
+}
+```
+
+The corresponding functions that deal with staking `Coin<SUI>` are `entry` functions and thus directly transfer the resulting `StakedSui` back to the sender.
+
+## Motivation
+
+**Transparency.** Without the `store` ability on `StakedSui`, projects building on top of the native Sui staking mechanics require a centralized address to store the `StakedSui` objects. This will limit the overall reach of these projects due to the high level of trust required for a user to feel comfortable in interacting with their product. In some cases this can also lead to a loss of funds, if the keys for the centralized account are lost or stolen.
+
+&nbsp;&nbsp;&nbsp;&nbsp; In order to enable truly trustless, transparent, and composable extensions of Sui staking, the `StakedSui` struct requires the `store` ability. This change moves the underlying, extrinsic trust in the project to the inherent safety in the relevant Sui Move modules. Trust can only be reassured, while smart contracts can go through many rounds of audits and tests to prove correctness.
+
+**Composability.** In order to allow the use of `StakedSui` flows in programmable transactions, `public`, non-`entry` variations of `request_add_stake` and `request_add_stake_mul_coins` are required that directly return the `StakedSui` objects. Without these functions, the inclusivity of Sui staking flows are limited within programmable txs.
+
+**Expressiveness.** By design, `StakedSui` is yield-bearing: every epoch an amount of `Coin<SUI>` rewards is distributed to each Validator's `StakingPool` to be later redeemed by unstaking a `StakedSui` position. A natural requirement for any protocol utilizing `StakedSui` is to determine the total value (i.e. principal + reward) accrued in the position. Currently, there exists a function to query the underlying principal value of a `StakedSui` object but not to query accrued rewards or the state of the associated `StakingPool`.
+
+&nbsp;&nbsp;&nbsp;&nbsp; To enable better on-chain interaction with `StakedSui` objects, additional `sui_system` getter functions are required.
+
+## Prerequisites
+
+In order to upgrade a package, your changes must satisfy a list of requirements. Adding new abilities to existing structs is strictly prohibited and thus, as it stands, **adding `store` to `StakedSui` will not pass the upgrade verification check**. To get around this, upgrades for _only_ the `sui_system` package should have a relaxed set of requirements.
```

Comment:
In general, we want to allow upgrades to add abilities to types, but doing so requires creating a new type identity in a way, so that objects have to explicitly opt in to being loaded with the new abilities. 

But this set of problems doesn't really pertain to the `sui` or `sui_system` packages, as they are upgraded "in-place" (upgraded mutably, without creating a second package). 

We can definitely allow adding abilities on types for these system packages (much sooner than we could for other package upgrades)

Reply:
@tnowacki, @emmazzz, hey, gods. When are you going to release this SIP6?
I think SUI liquid staking will bring in more users to sui ecosystem.